tests/flow_counter/flow_counter_utils.py

```diff
@@ -0,0 +1,278 @@
+            result, message = verify_route_flow_counter_stats(self.expected_stats, actual_stats)
+            if not result:
+                return result, message
+
+        if len(self.expected_stats) == 1 and len(self.route_pattern_list) == 1:
+            logger.info('Checking route flow counter stats after clearing by pattern')
+            clear_route_flow_counter_by_pattern(self.dut, self.route_pattern_list[0])
+        else:
+            logger.info('Checking route flow counter stats after clearing all routes')
+            clear_route_flow_counter(self.dut)
+        for prefix, value in self.expected_stats.items():
+            for key in value:
+                self.expected_stats[prefix][key] = '0'
+
+        actual_stats = parse_route_flow_counter_stats(self.dut)
+        return verify_route_flow_counter_stats(self.expected_stats, actual_stats)
+
+
+def is_route_flow_counter_supported(dut):
+    """Check if route flow counter is supported on this platform
+
+    Args:
+        dut (object): DUT object
+
+    Returns:
+        bool: True if supported
+    """
+    skip, _ = check_skip_release(dut, skip_versions)
+    if skip:
+        return False
+
+    global support_route_flow_counter
+    if support_route_flow_counter is None:
+        if not wait_until(CAPABILITY_WAIT_TIME_IN_SEC, CAPABILITY_CHECK_INTERVAL_IN_SEC, 0, get_route_flow_counter_capability, dut):
+            support_route_flow_counter = False
+            pytest_assert(False, 'Failed to get route flow counter capability')
+        if not support_route_flow_counter:
+            logger.info('Route flow counter is not supported on this platform')
+    return support_route_flow_counter
+
+
+def get_route_flow_counter_capability(dut):
+    """Get route flow counter capability from STATE DB
+
+    Args:
+        dut (object): DUT object
+
+    Returns:
+        bool: True if capability is successfully retrieved from STATE DB
+    """
+    global support_route_flow_counter
+    support = dut.shell('sudo sonic-db-cli STATE_DB HGET "FLOW_COUNTER_CAPABILITY_TABLE|route" support')['stdout'].strip()
+    if support == 'true':
+        support_route_flow_counter = True
+    elif support == 'false':
+        support_route_flow_counter = False
+    return support_route_flow_counter is not None
+
+
+def set_route_flow_counter_status(dut, status):
+    """Set route flow counter status
+
+    Args:
+        dut (object): DUT object
+        status (bool): Enable if True else disable
+    """
+    dut.command('counterpoll flowcnt-route {}'.format('enable' if status else 'disable'))
+
+
+def set_route_flow_counter_interval(dut, interval):
+    """Set route flow counter interval
+
+    Args:
+        dut (object): DUT object
+        interval (int): Query interval value in ms
+    """
+    dut.command('counterpoll flowcnt-route interval {}'.format(interval))
+
+
+def set_route_flow_counter_pattern(dut, route_pattern, max_match_count=30):
```

Comment:
To me it seems more readable to get another parameter- VRF, rather than encapsulating it in the route pattern string

Reply:
Indeed, using VRF and prefix here makes this function more readable. However, I would like to keep this based on two reasons:
1. Most of case VRF is none(default VRF)
2. If we use VRF and prefix, the caller has to use a tuple (<vrf>, <prefix>) to represent the route pattern, it increases unneeded complexity of caller side. And a tuple is not so readable.